### PR TITLE
packagekit: Polish the case when there are only security fixes

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
I just got "9 updates, including 9 security fixes", and I was thinking "okay I see the security fixes, where is the rest?"  Math is hard, I know, but.... :-)

**Comment by mvollmer** [comment]
Should we also do something about the "install" buttons?  They both do exactly the same thing when there are only security fixes?

File: pkg/packagekit/updates.jsx [hunk 1]
```diff
@@ -176,12 +176,16 @@ function HeaderBar(props) {
...
 
     if (props.state == "available") {
-        state = cockpit.ngettext("$0 update", "$0 updates", num_updates);
         for (let u in props.updates)
             if (props.updates[u].security)
                 ++num_security;
-        if (num_security > 0)
-            state += cockpit.ngettext(", including $1 security fix", ", including $1 security fixes",  num_security);
+        if (num_updates == num_security)
+            state = cockpit.ngettext("$1 security fix", "$1 security fixes", num_security);
```
Comment (martinpitt) [inline pkg/packagekit/updates.jsx]: That branch would also hit for 0 updates, wouldn't it? That would look a bit odd. I think an empty string would be fine in this case, the big "your system is up to date" empty state pattern is enough information.

Comment (mvollmer) [reply pkg/packagekit/updates.jsx]: With 0 updates, the state isn't "available", so we don't end up here at all.

Comment (martinpitt) [reply pkg/packagekit/updates.jsx]: Right, sorry about that. Seems I already dumped all my working knowledge about this from my brain :-)

**Comment by mvollmer** [comment]
> It should then also get a test case, it's simple enough to set up.

Done.

> As for the buttons, I think it would be fine to only show the "install security updates" button in this case and make it primary.

Done.

File: pkg/packagekit/updates.jsx [hunk 2]
```diff
@@ -800,12 +810,16 @@ class OsUpdates extends React.Component {
...
+
                     applyAll = (
                         <button className="btn btn-primary" onClick={ () => this.applyUpdates(false) }>
-                            {_("Install All Updates")}
+                            { num_updates == num_security_updates?
+                              _("Install Security Updates") : _("Install All Updates") }
                         </button>);
 
-                    if (this.state.haveSecurity) {
+                    if (num_updates > num_security_updates) {
```
Comment (martinpitt) [inline pkg/packagekit/updates.jsx]: I think this still requires a check that `num_security_updates > 0`, to replace the previous `haveSecurity`. We don't want to show that button if there are only "normal" updates available.

Comment (mvollmer) [reply pkg/packagekit/updates.jsx]: Indeed, ouch.  Fixed.

**Pull request merged**